File: javascript/6_kyu/detect_pangram.js
// Detect Pangram
// https://www.codewars.com/kata/545cedaa9943f7fe7b000048/train/javascript

function isPangram(text) {
    var alpha = "a b c d e f g h i j k l m n o p q r s t u v w x y z".split(" ")

    for (var i = 0; i < text.length; i++) {
        var text = text.toLowerCase();
        if (alpha.includes(text[i])) {
            alpha.splice(alpha.indexOf(text[i]), 1);
        }
    }

    return alpha.length == 0;
}

console.log(isPangram("The quick brown fox jumps over the lazy dog."));
console.log(isPangram("This is not a ! Pangram."));
console.log(isPangram("Pack my box with five dozen liquor jugs."));
console.log(isPangram("ABCD45EFGH,IJK,LMNOPQR56STUVW3XYZ"));

File: python/6_kyu/detect_pangram.py
# Detect Pangram
# https://www.codewars.com/kata/545cedaa9943f7fe7b000048/train/python

def is_pangram(text):
    alpha = "a b c d e f g h i j k l m n o p q r s t u v w x y z".split(" ")

    text = [i.lower() for i in text if i.isalpha()]

    for i in text:
        if i in alpha:
            alpha.remove(i)
    
    return len(alpha) == 0

print(is_pangram("The quick, brown fox jumps over the lazy dog"))
print(is_pangram("This is not a pangram!"))